monai/transforms/rand_rotation.py

```diff
@@ -0,0 +1,210 @@
+            return data
+    ```
+    """
+
+    def __init__(self, keys):
+        self.keys = keys if isinstance(keys, (list, tuple)) else (keys,)
+        if not self.keys:
+            raise ValueError('keys unspecified')
+        for key in self.keys:
+            if not isinstance(key, Hashable):
+                raise ValueError('keys should be a hashable or a list of hashables, got {}'.format(type(key)))
+
+
+@export
+class Rotate90(Transform):
+    """
+    Rotate an array by 90 degrees in the plane specified by `axes`.
+    """
+
+    def __init__(self, k=1, axes=(1, 2)):
+        """
+        Args:
+            k (int): number of times to rotate by 90 degrees.
+            axes (2 ints): defines the plane to rotate with 2 axes.
+        """
+        self.k = k
+        self.plane_axes = axes
+
+    def __call__(self, img):
+        return np.rot90(img, self.k, self.plane_axes)
+
+
+@export
+class Rotate90d(MapTransform):
+    """
+    dictionary-based wrapper of Rotate90.
+    """
+
+    def __init__(self, keys, k=1, axes=(1, 2)):
+        """
+        Args:
+            k (int): number of times to rotate by 90 degrees.
+            axes (2 ints): defines the plane to rotate with 2 axes.
+        """
+        MapTransform.__init__(self, keys)
+        self.k = k
+        self.plane_axes = axes
+
+    def __call__(self, data):
+        d = copy(data)
+        for key in self.keys:
+            if key in d:
+                d[key] = Rotate90(self.k, self.plane_axes)(d[key])
+            else:
+                raise KeyError('data dict does contain {} key'.format(key))
+        return d
+
+
+@export
+class RandRotate90(Randomizable, MapTransform):
+    """
+    With probability `prob`, input arrays are rotated by 90 degrees
+    in the plane specified by `axes`.
+    """
+
+    def __init__(self, keys, prob=0.1, max_k=3, axes=(1, 2)):
+        """
+        Args:
+            keys (hashable items): keys of the corresponding items to be transformed.
+                see also: monai.transform.rand_rotation.MapTransform
+            prob (float): probability of rotating.
+                (Default 0.1, with 10% probability it returns a rotated array.)
+            max_k (int): number of rotations will be sampled from `np.random.randint(max_k) + 1`.
+                (Default 3)
+            axes (2 ints): defines the plane to rotate with 2 axes.
+        """
+        MapTransform.__init__(self, keys)
+
+        self.prob = min(max(prob, 0.0), 1.0)
+        self.max_k = max_k
+        self.axes = axes
+
+        self._do_transform = False
+        self._rand_k = 0
+
+    def randomise(self):
+        self._rand_k = self.R.randint(self.max_k) + 1
+        self._do_transform = self.R.random() < self.prob
+
+    def __call__(self, data):
+        self.randomise()
+        if not self._do_transform:
+            return data
+        return Rotate90d(self.keys, self._rand_k, self.axes)(data)
```

Comment:
Same idea, there should be a member instance of `Rotate90d` (say `rotator`) then this line would be `return self.rotator(data)`.

Reply:
`self._rand_k` is dynamic here so I couldn't make an instance beforehand...